### ISSUE TITLE
`emit-translation-events` CLI flag to suppress event creation

### ISSUE THREAD

**Issue by mflendrich** [Kong/kubernetes-ingress-controller#5252]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Problem Statement

Today it is not possible to suppress creation of k8s events for translation successes/failures.
Those events can be a source of load on k8s API that is higher than necessary (#4127) and can be unwanted (e.g. in an emergency where the load on the API server is impacting the cluster).

### Proposed Solution

Add a config knob to disable KIC's module that creates events.

There is prior art: `update-status`

https://github.com/Kong/kubernetes-ingress-controller/blob/cff127d03acf818a3865ef6d8ee016ffd1f15087/internal/manager/config.go#L229C42-L230

The idea is that we add a new flag alongside `update-status` (proposed name: `emit-translation-events`) defaulting to true.

### Additional information

Related to #4127. #4127 is the long-term solution that, if #4127 is enabled by default, the likelihood of translation events causing trouble `emit-translation-events` flag should be way less.

### Acceptance Criteria

- [ ] `emit-translation-events` flag implemented, set as default=true, if disabled: KIC creates no events in the Kubernetes API.

**Comment by czeslavo** [comment]
I suggest we rename the flag to `--emit-kubernetes-events` and use it to determine whether we should emit any type of Events (not only `KongConfigurationTranslationFailed`, but also `KongConfigurationSucceeded` and `KongConfigurationApplyFailed`).

**Comment by rainest** [comment]
tl;dr from https://github.com/Kong/kubernetes-ingress-controller/issues/4127#issuecomment-1832944974 is that we should already be limited to 1 Event update per failure per 300s after an initial ramp up period. The exact rate is tunable, though documentation is sparse and the the limiter behavior varies depending on your Even emission patterns.

There is a corner case at startup where, if you've let a large quantity N of broken resources accumulate, you would see N Event updates every 3s until you hit the 25 Event duplicate count before throttling starts.

**Comment by randmonkey** [comment]
Reopen since the backport to 2.12.x and 3.0.x is not done.

**Comment by czeslavo** [comment]
Posted a proposal PR (https://github.com/Kong/kubernetes-ingress-controller/pull/5299) to rename the flag to `--emit-kubernetes-events` (because of [comment](https://github.com/Kong/kubernetes-ingress-controller/issues/5252#issuecomment-1831692491) reasons).

**Comment by tao12345666333** [comment]
Already backported.
Let's close this one
